docs/GlutenUsage.md

```diff
@@ -15,8 +15,7 @@ Based on the different environment, there are some parameters can be set via -D
 | build_velox                                           | Enable or Disable building the CPP part of Velox backend                                                                                                                           | OFF                                                |
```

Comment:
can this be enabled once define backends-velox? 
Let's clean the parameters a bit. Now we have 

backends-velox
compile_velox
velox_home 
build_velox_from_source    --> can it be implied by velox_home and compile_velox?
build_velox  --> can this be implied by compile_velox?
velox_build_type  --> can we search release or debug or relwithdebinfo automatically? It's rarelly customer specify another folder

Reply:
1. will remove build_velox_from_source, use velox_home and compile_velox to imply.
2. will remove build_velox, use backends-velox to imply it.
3. velox_build_type is removed in this PR. And we can use build_type. build_type can be release or debug or relwithdebinfo now, and gluten can search corresponding version of velox now.

4. This PR also achieve only build velox but not third party lib, So the next compile progress will be faster.

Reply:
> 2. I think build_velox can not be remove. it's different from compile_velox now. build_velox decide whether we need build gluten with velox.  And compile_velox means whether we need recompile_velox if user specify the velox_home.

If we specify backends-velox, should we must build gluten with velox? 

> 3. velox_build_type can be release or debug or relwithdebinfo now, and gluten can search corresponding version of velox  automatically now.

Then let's remove the config in case of confusing. We can put a node in velox_home and say script will search release/debug or rewithdebinfo in velox_home. Also if it's not searched, let's print the hint there.

> 4. This PR also achieve only build velox but not third party lib, So the next compile progress will be faster.

what should we set to achieve this?

Reply:
I remember that `build_velox` stands for "to build gluten c++ velox-backend code", while `build_velox_from_source` stands for "to download and build velox from source"

Although the two were very confusing to me. IMHO we may be able to have a suffix for backend build option, e.g.

`build_velox` -> `build_velox_backend`
so the rule should apply to other BEs
`build_gazelle_cpp` -> `build_gazelle_cpp_backend`

Reply:
I think so.

Reply:
If we need the feature, Let's document it clearly